fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -720,6 +687,32 @@ public Node getNodeVersion(final String label) {
 
     }
 
+    /**
+     * Helps ensure that there are no terminating slashes in the predicate.
+     * A terminating slash means ModeShape has trouble extracting the localName, e.g., for
+     * http://myurl.org/.
+     *
+     * @see <a href="https://jira.duraspace.org/browse/FCREPO-1409"> FCREPO-1409 </a> for details.
+     */
+    private Collection<IllegalArgumentException> checkInvalidPredicates(final UpdateRequest request) {
+        return request.getOperations().stream()
+                .flatMap(x -> {
+                    if (x instanceof UpdateModify) {
+                        final UpdateModify y = (UpdateModify)x;
+                        return Stream.concat(y.getInsertQuads().stream(), y.getDeleteQuads().stream());
+                    } else if (x instanceof UpdateDataInsert || x instanceof UpdateDataDelete) {
+                        return ((UpdateData)x).getQuads().stream();
+                    } else if (x instanceof UpdateDeleteWhere) {
+                        return ((UpdateDeleteWhere)x).getQuads().stream();
+                    } else {
+                        return Stream.empty();
+                    }
```

Comment:
This whole thing (above) might look better without all the `instanceof` instead using `Update::visit`. It would get rid of all the casting, too, and the code would be appropriately sensitive to Jena introducing any new subtypes of `Update`.